blocks/http/include/gnuradio-4.0/http/HttpBlock.hpp

```diff
@@ -184,56 +197,63 @@ class HttpBlock : public gr::Block<HttpBlock<T>, BlockingIO<false>, HttpBlockDoc
     }
 
     void
-    runThread() {
+    startThread() {
+        if (_thread) {
+            _thread.reset();
```

Comment:
@RalphSteinhagen What is this new check and reset for? This is effectively a noop, as the line immediately below it assigns a new `shared_ptr`, which includes implicitly calling `reset()` on the old one, or am I missing an edge case here?

Reply:
Yes, there is a data-race in the reset/thread destruction. The new thread creation and deletion of the old access shared resources which isn't safe. Forcing the reset mitigated this.